sdk/src/main/scala/com/horizen/api/http/SidechainTransactionApiRoute.scala

```diff
@@ -447,6 +446,107 @@ case class SidechainTransactionApiRoute(override val settings: RESTApiSettings,
+    entity(as[ReqOpenStake]) { body =>
+      buildOpenStakeTransaction(body) match {
+        case Success(tx) =>
+          if (body.automaticSend.getOrElse(true)) {
+            validateAndSendTransaction(tx.asInstanceOf[SidechainTypes#SCBT])
+          } else {
+            if (body.format.getOrElse(false))
+              ApiResponseUtil.toResponse(TransactionDTO(tx.asInstanceOf[SCBT]))
+            else
+              ApiResponseUtil.toResponse(TransactionBytesDTO(BytesUtils.toHexString(companion.toBytes(tx.asInstanceOf[SCBT]))))
+          }
+        case Failure(exception) =>
+          ApiResponseUtil.toResponse(GenericTransactionError("GenericTransactionError", JOptional.of(exception)))
+      }
+    }
+  }
+
+  def createOpenStakeTransactionSimplified: Route = (post & path("createOpenStakeTransactionSimplified")) {
+    entity(as[ReqOpenStakeSimplified]) { body =>
+     buildOpenStakeTransactionSimplified(body) match {
+       case Success(tx) =>
+         if (body.automaticSend.getOrElse(true)) {
+           validateAndSendTransaction(tx.asInstanceOf[SidechainTypes#SCBT])
+         } else {
+           if (body.format.getOrElse(false))
+             ApiResponseUtil.toResponse(TransactionDTO(tx.asInstanceOf[SCBT]))
+           else
+             ApiResponseUtil.toResponse(TransactionBytesDTO(BytesUtils.toHexString(companion.toBytes(tx.asInstanceOf[SCBT]))))
+         }
+       case Failure(exception) =>
+         ApiResponseUtil.toResponse(GenericTransactionError("GenericTransactionError", JOptional.of(exception)))     }
+    }
+  }
+
+  private def buildOpenStakeTransactionSimplified(body: ReqOpenStakeSimplified): Try[OpenStakeTransaction] = {
+    applyOnNodeView { sidechainNodeView =>
+      val wallet = sidechainNodeView.getNodeWallet
+
+      //Collect input box
+      val inputBox = wallet.allBoxes().asScala
+        .filter(box => box.proposition().equals(PublicKey25519PropositionSerializer.getSerializer.parseBytes(BytesUtils.fromHexString(body.forgerProposition))))
+      if (inputBox.isEmpty)
+        throw new IllegalArgumentException("Unable to find input for Proposition "+body.forgerProposition)
+      createAndSignOpenStakeTransaction(wallet, inputBox.head, body.forgerProposition, body.forgerListIndex, body.fee)
+    }
+  }
+
+  /**
+   * Create and sign a openStakeTransaction
+   * @return a signed OpenStakeTransaction or a GenericTransactionError
+   */
+  private def buildOpenStakeTransaction(body: ReqOpenStake): Try[OpenStakeTransaction] = {
+    applyOnNodeView { sidechainNodeView =>
+      val wallet = sidechainNodeView.getNodeWallet
+
+      //Collect input box
+      val inputBox = wallet.allBoxes().asScala
+        .filter(box => BytesUtils.toHexString(box.id()).equals(body.transactionInput.boxId))
+      if (inputBox.isEmpty) {
+        throw new IllegalArgumentException("Unable to find input!")
+      }
+      if (inputBox.length > 1) {
```

Comment:
with `find` we will get `option` type as a result, so this check can be removed.

Reply:
Done